### PR TITLE
feat: add `make k8s-lint` target and a `make k8s-crd-extract` target

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

**Review by russellb — CHANGES_REQUESTED** [review]

File: Makefile [hunk 1]
```diff
@@ -121,6 +121,39 @@ dist/.yaml-lint: $(wildcard */**/*.yaml) | gen-docs dist
 	$(CMD_PREFIX) yamllint -c .yamllint.yaml deploy --strict
 	$(CMD_PREFIX) touch $@
 
+
+.PHONY: k8s-lint
+k8s-lint: dist/.k8s-lint ## Lint the kubernetes deployment files
```
[the remark this answers is not in the document]
Comment (chirino) [reply Makefile:126]: done

File: Makefile [hunk 2]
```diff
@@ -121,6 +121,39 @@ dist/.yaml-lint: $(wildcard */**/*.yaml) | gen-docs dist
...
+
+dist/.k8s-lint: $(shell find deploy -type f ) | dist
```
Comment (russellb) [inline Makefile:128]: Missing a `touch $@` in here

**Review by russellb — CHANGES_REQUESTED** [review]
one more little thing, missed it last time, sorry

File: Makefile (hunk 2 above)
Comment (chirino) [reply Makefile:128]: good catch.  fixed.

**Review by russellb — APPROVED** [review]

**Pull request merged**